ext/crypto/00_crypto.js

```diff
@@ -898,7 +1112,274 @@
+              }
+
+              // 7.
+              if (jwk.alg) {
+                switch (jwk.alg) {
+                  case "PS1": {
+                    hash = "SHA-1";
+                    break;
+                  }
+                  case "PS256": {
+                    hash = "SHA-256";
+                    break;
+                  }
+                  case "PS384": {
+                    hash = "SHA-384";
+                    break;
+                  }
+                  case "PS512": {
+                    hash = "SHA-512";
+                    break;
+                  }
+                  default:
+                    throw new TypeError("unreachable");
+                }
+              }
+
+              //8.
+              if (hash) {
+                const normalizedHash = normalizeAlgorithm(hash, "digest");
+
+                if (normalizedHash.name !== normalizedAlgorithm.hash.name) {
+                  throw new DOMException(
+                    "Mismatched hash algorithm for JWK.alg",
+                    "DataError",
+                  );
+                }
+              }
+
+              modulusLength = decodeSymmetricKey(jwk.n).length * 8;
+
+              //9.
+              if (jwk.d) {
+                // Section 6.3.2 of RFC7518
+                if (!jwk.e || !jwk.n) {
+                  throw new DOMException(
+                    "Missing n or e components in RSA key",
+                    "DataError",
+                  );
+                }
+
+                type = "private";
+                data = buildRsaPkcs({
+                  n: decodeSymmetricKey(jwk.n),
+                  e: decodeSymmetricKey(jwk.e),
+                  d: decodeSymmetricKey(jwk.d),
+                  p: decodeSymmetricKey(jwk.p),
+                  q: decodeSymmetricKey(jwk.q),
+                  dp: decodeSymmetricKey(jwk.dp),
+                  dq: decodeSymmetricKey(jwk.dq),
+                  qi: decodeSymmetricKey(jwk.qi),
+                }, "pkcs8");
+              } else {
+                // Section 6.3.1 of RFC7518
+                if (!jwk.e || !jwk.n) {
+                  throw new DOMException(
+                    "Missing n or e components in RSA key",
+                    "DataError",
+                  );
+                }
+
+                type = "public";
+                data = buildRsaPkcs({
+                  n: decodeSymmetricKey(jwk.n),
+                  e: decodeSymmetricKey(jwk.e),
+                }, "spki");
+              }
+
+              break;
+            }
+            case "spki": {
+              // 1.
+              if (
+                ArrayPrototypeFind(
+                  keyUsages,
+                  (u) => !ArrayPrototypeIncludes(["verify"], u),
+                ) !== undefined
+              ) {
+                throw new DOMException("Invalid key usages", "SyntaxError");
+              }
+              //2.
+              const keyInfo = parseRsaPkcs(keyData, "spki");
```

Comment:
Use the `spki` crate for this. https://docs.rs/spki/0.4.0/spki/

Reply:
Ok. I think its best to wait on #11891, then include spki as an option in op_crypto_import_key.

Reply:
@littledivy - I'll create a new branch/PR for spki, converting in rust to a pkcs1 public key. Can probably include export as well .. and was wondering what are your thoughts on exportKey for spki (and jwk), since we'll need to parse the raw pkcs1 to extract the key components. Rust or js?

Going forward, I saw some comments in `op_crypto_import_key` about importing jwk .. is this on the to-do list?

Reply:
@SeanWykes Preferably using Rust since we already have the dependencies. See https://github.com/denoland/deno/pull/12022#discussion_r710707372 

> Going forward, I saw some comments in op_crypto_import_key about importing jwk .. is this on the to-do list?

It's on the todo but feel free to work on it!

Reply:
I've opened a PR for exporting spki keys https://github.com/denoland/deno/pull/12114. Hopefully that can help you with importKey